yeti_snowplow/src/obstacle_reaction.cpp

```diff
@@ -1,29 +1,198 @@
-#include "ros/ros.h"
-#include "yeti_snowplow/obstacles.h"
-//#include "yeti_snowplow/robot_position.h"
-#include "yeti_snowplow/turn.h"
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+#include <geometry_msgs/Pose2D.h>
+#include <sensor_msgs/PointCloud.h>
+#include <yeti_snowplow/obstacles.h>
+#include <yeti_snowplow/obstacle.h>
+#include <yeti_snowplow/location_point.h>
+#include <yeti_snowplow/lidar_point.h>
+#include <yeti_snowplow/turn.h>
+#include <yeti_snowplow/target.h>
+#include <sensor_msgs/LaserScan.h>
+#include <sensor_msgs/PointCloud.h>
+#include <laser_geometry/laser_geometry.h>
+#include <yeti_snowplow/waypoint.h>
+#include "buffer.h"
+#include <math.h>
+#include <vector>
+#include <algorithm>
 
-ros::Publisher turnPub;
+class ObstacleReaction
+{
+public:
+	// void obstaclePositionsCallback(const yeti_snowplow::obstacles::ConstPtr& groupOfObstacles)
+	// {
+	// 	obstacles = groupOfObstacles->obstacles;
+	// }
+	ObstacleReaction()
+	{
+		wayPointID = 0;
 
-void obstaclePositionsCallback(const yeti_snowplow::obstacles::ConstPtr& obstaclePositions){
-	
-}
+		//turnPub = n.advertise<yeti_snowplow::turn>("obstacle_reaction_turn", 1000);
+		//obstaclePositionsSub = n.subscribe("obstacles", 1000, obstaclePositionsCallback);
+		robotPositionSub = n.subscribe("robot_position", 1000, &ObstacleReaction::robotPositionCallback, this);
+		waypointClient = n.serviceClient<yeti_snowplow::waypoint>("waypoint");
+		scanSub = n.subscribe("obstacles", 1000, &ObstacleReaction::scanCallback,this);
```

Comment:
This should be /obstacle_detection/obstacles. The Obstacle Detection node also needs to be fixed to have this.